examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml

```diff
@@ -0,0 +1,175 @@
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: argo-hub.image-test-template.0.0.1
+  annotations:
+    argo-hub/version: '0.0.1'
+    argo-hub/description: 'Workflow template for a deployment that builds and tests an image through service containers'
+    argo-hub/categories: 'argo'
+    argo-hub/license: 'MIT'
+    argo-hub/owner_name: 'Matthew Chung'
+    argo-hub/owner_email: 'matthew.chung@codefresh.io'
+    argo-hub/owner_avatar: 'https://github.com/matthewchungcodefresh.png'
+    argo-hub/owner_url: 'https://github.com/matthewchungcodefresh'
+    argo-hub/icon_url: "https://github.com/codefresh-io/argo-hub/blob/main/workflows/image-test-workflow/assets/icon.svg"
+    argo-hub/icon_background: "#f4f4f4"
+spec:
+  volumes:
+    - name: docker-config
+      secret:
+        items:
+          - key: .dockerconfigjson
+            path: config.json
+        secretName: '{{ inputs.parameters.DOCKER_CONFIG_SECRET }}'
+  entrypoint: main
+  templates:
+    - name: main
+      metadata:
+        annotations:
+          argo-hub-template/description: 'Builds a docker image using Kaniko and tests the image through service containers'
+          argo-hub-template/icon_url: "https://github.com/codefresh-io/argo-hub/blob/main/workflows/image-test-workflow/assets/icon.svg"
+          argo-hub-template/icon_background: "#f4f4f4"
+      inputs:
+        parameters:
+          # Create a docker-registry secret with your docker credentials and apply it to your cluster under the runtime namespace
+          # https://jamesdefabia.github.io/docs/user-guide/kubectl/kubectl_create_secret_docker-registry/
+          - name: DOCKER_CONFIG_SECRET
+            default: docker-config  # the name of the registry-creds secret you created
+          # Create MYSQL secret and apply it to your cluster under the runtime namespace
+          #   kubectl create secret generic mysql-secret \
+          #   --from-literal=root-password=$MYSQL_ROOT_PASSWORD --dry-run=client \
+          #   --from-literal=user=$MYSQL_USER --dry-run=client \
+          #   --from-literal=password=$MYSQL_PASSWORD --dry-run=client \
+          #   --save-config -o yaml | kubectl apply -f - -n $NAMESPACE
+          - name: MYSQL_SECRET
+            default: mysql-secret
+          - name: MYSQL_ROOT_PASSWORD
```

Comment:
those are not required at the top template as they are values from the secret

Reply:
I see.
those are the keys to extract value  from the secret
just keep them hardcoded at line 147, 152 and 157 to limit the numbers of parameters